plugins/regex_revalidate/regex_revalidate.cc

```diff
@@ -0,0 +1,757 @@
+  return 0;
+}
+
+} // namespace
+
+TSRemapStatus
+TSRemapDoRemap(void *ih, TSHttpTxn txnp, TSRemapRequestInfo *rri)
+{
+  TSCont const main_cont = TSContCreate(main_handler, nullptr);
+
+  TSContDataSet(main_cont, ih);
+  TSHttpTxnHookAdd(txnp, TS_HTTP_CACHE_LOOKUP_COMPLETE_HOOK, main_cont);
+  TSHttpTxnHookAdd(txnp, TS_HTTP_TXN_CLOSE_HOOK, main_cont);
+
+  return TSREMAP_NO_REMAP;
+}
+
+TSReturnCode
+TSRemapNewInstance(int argc, char *argv[], void **ih, char *errbuf, int errbuf_size)
+{
+  TSDebug(LOG_PREFIX, "Starting remap init");
+  PluginState *const pstatenew = new PluginState;
+
+  if (!pstatenew->fromArgs(argc - 1, (char const **)(argv + 1))) {
+    TSError("[regex_revalidate] Remap plugin registration failed");
+    delete pstatenew;
+    return TS_ERROR;
+  }
+
+  pstatenew->remap_from = argv[0];
+  pstatenew->remap_to   = argv[1];
+
+  *ih = static_cast<void *>(pstatenew);
+
+  TSAssert(nullptr != config_cont);
+
+  // put this plugin on the config scanner
+  TSMutex const config_mutex = TSContMutexGet(config_cont);
+  TSMutexLock(config_mutex);
+
+  // pass epoch state after a config reload
+  for (PluginState *const pstate : plugin_states) {
+    if (pstatenew->isSameRemap(*pstate)) {
+      TSDebug(LOG_PREFIX, "Transferring active rules across reload");
+      pstatenew->invalidate_vec = pstate->invalidate_vec;
+      break;
+    }
+  }
+
+  plugin_states.emplace_back(pstatenew);
+  time_t const timenow = time(nullptr);
+  if (pstatenew->tryConfigReload(timenow)) {
+    pstatenew->logConfig();
+  }
+  pstatenew->prune(timenow);
+
+  TSMutexUnlock(config_mutex);
+
+  TSDebug(LOG_PREFIX, "Remap plugin registration succeeded");
+
+  return TS_SUCCESS;
+}
+
+void
+TSRemapDeleteInstance(void *ih)
+{
+  TSDebug(LOG_PREFIX, "Deleting remap plugin instance");
+
+  TSAssert(nullptr != ih);
+
+  if (nullptr != ih) {
+    PluginState *const pstate = static_cast<PluginState *>(ih);
+
+    TSAssert(nullptr != config_cont);
+
+    // remove the pstate from the plugin states
+    TSMutex const config_mutex = TSContMutexGet(config_cont);
+    TSMutexLock(config_mutex);
+
+    std::vector<PluginState *>::iterator const itf = std::find(plugin_states.begin(), plugin_states.end(), pstate);
+
+    if (plugin_states.end() != itf) {
+      std::swap(*itf, plugin_states.back());
+      plugin_states.pop_back();
+    } else {
+      TSError("Unable to find registered plugin config state");
+    }
+
+    TSMutexUnlock(config_mutex);
+    delete pstate;
+  }
+}
+
+namespace
+{
+bool
+check_ts_version()
+{
+  const char *ts_version = TSTrafficServerVersionGet();
+
+  if (ts_version) {
+    int major_ts_version = 0;
+    int minor_ts_version = 0;
+    int micro_ts_version = 0;
+
+    if (sscanf(ts_version, "%d.%d.%d", &major_ts_version, &minor_ts_version, &micro_ts_version) != 3) {
+      return false;
+    }
+
+    if ((TS_VERSION_MAJOR == major_ts_version)) {
+      return true;
+    }
+  }
+
+  return false;
+}
```

Comment:
Why not just:
```
bool 
check_ts_version() { return TS_VERSION_MAJOR == TSTrafficServerVersionGetMajor(); }
```
?

Reply:
probably some hacked up copy/paste from another plugin.

Reply:
I suggested a fairly small change to make this check automatic for all plugins:  https://github.com/apache/trafficserver/pull/4523 .  But it was vaporized by apocalyptic bikeshedding.